search/api.py

```diff
@@ -161,40 +162,49 @@ def _apply_general_query_filters(search, user):
         else []
     )
 
+    # Search for comments and posts from channels
     channels_filter = Q(
         "terms", channel_type=[CHANNEL_TYPE_PUBLIC, CHANNEL_TYPE_RESTRICTED]
     ) | ~Q("terms", object_type=[COMMENT_TYPE, POST_TYPE])
 
+    # Exclude deleted comments and posts
     content_filter = (Q("term", deleted=False) & Q("term", removed=False)) | ~Q(
         "terms", object_type=[COMMENT_TYPE, POST_TYPE]
     )
 
+    # Search only published courses
     course_filter = Q("term", published=True) | ~Q("terms", object_type=[COURSE_TYPE])
 
+    # Search only published bootcamps
     bootcamp_filter = Q("term", published=True) | ~Q(
         "terms", object_type=[BOOTCAMP_TYPE]
     )
 
+    # Search programs
     program_filter = ~Q("terms", object_type=[PROGRAM_TYPE])
 
-    user_list_filter = (
-        Q("term", privacy_level=PrivacyLevel.public.value)
-        | Q("term", author_id=user.id)
-        | ~Q("terms", object_type=[USER_LIST_TYPE])
-    )
-
+    # Search public channels and channels user is a contributor/moderator of
     if channel_names:
         channels_filter = channels_filter | Q("terms", channel_name=channel_names)
 
-    return (
+    filters = (
         search.filter(channels_filter)
         .filter(content_filter)
         .filter(course_filter)
         .filter(bootcamp_filter)
         .filter(program_filter)
-        .filter(user_list_filter)
     )
 
+    if not user.is_anonymous:
+        # Search the user's lists
+        user_list_filter = (
```

Comment:
Should this actually be:

```python
user_list_filter = (
    Q("term", privacy_level=PrivacyLevel.public.value)
    | ~Q("terms", object_type=[USER_LIST_TYPE])
)
if not user.is_anonymous:
    user_list_filter = user_list_filter | Q("term", author_id=user.id)
```

and can then be moved back to its original location so that public lists are still searchable but private lists are only searchable if logged in?

Reply:
Right, I misread the query as just returning the user's own lists, which would be irrelevant if anonymous, will change.